cmd/pitr/collector/collector.go

```diff
@@ -242,35 +242,53 @@ func (c *Collector) CollectBinLogs(ctx context.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "get GTID sets")
 	}
-	var sourceID string
-	for i := len(list) - 1; i >= 0 && sourceID == ""; i-- {
-		sourceID = strings.Split(list[i].GTIDSet, ":")[0]
+	var lastGTIDSetList []string
+	for i := len(list) - 1; i >= 0 && len(lastGTIDSetList) == 0; i-- {
+		gtidSetList := list[i].GTIDSet.List()
+		if gtidSetList == nil {
+			continue
+		}
+		lastGTIDSetList = gtidSetList
 	}
 
-	if sourceID == "" {
+	if len(lastGTIDSetList) == 0 {
 		log.Println("No binlogs to upload")
 		return nil
 	}
 
-	c.lastSet, err = c.lastGTIDSet(ctx, sourceID)
-	if err != nil {
-		return errors.Wrap(err, "get last uploaded gtid set")
+	for _, gtidSet := range lastGTIDSetList {
+		sourceID := strings.Split(gtidSet, ":")[0]
+		c.lastUploadedSet, err = c.lastGTIDSet(ctx, sourceID)
+		if err != nil {
+			return errors.Wrap(err, "get last uploaded gtid set")
+		}
+		if !c.lastUploadedSet.IsEmpty() {
+			break
+		}
 	}
 
 	lastUploadedBinlogName := ""
 
-	if c.lastSet != "" {
-		for i := len(list) - 1; i >= 0; i-- {
-			if list[i].GTIDSet == c.lastSet {
-				lastUploadedBinlogName = list[i].Name
-				break
+	if !c.lastUploadedSet.IsEmpty() {
+		for i := len(list) - 1; i >= 0 && lastUploadedBinlogName == ""; i-- {
+			for _, gtidSet := range list[i].GTIDSet.List() {
+				for _, lastUploaded := range c.lastUploadedSet.List() {
+					isSubset, err := c.db.GTIDSubset(ctx, lastUploaded, gtidSet)
+					if err != nil {
+						return errors.Wrap(err, "check if gtid set is subset")
+					}
+					if isSubset {
+						lastUploadedBinlogName = list[i].Name
+						break
```

Comment:
this only break innermost for loop, do we want to continue if we found the binlog?

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1441/commits/8ce508d0bc168996f55fd5939dadab49bdf1b2e0